### PR TITLE
ci: add warning to actions with writeable GITHUB_TOKEN

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: .github/workflows/backport.yml [hunk 1]
```diff
@@ -2,6 +2,12 @@ name: Backport
 on:
   pull_request_target:
     types: [closed, labeled]
+
+# WARNING:
+# When extending this action, be aware that $GITHUB_TOKEN allows write access to
+# the github repository. This means that is should not evaluate user input in a
```
Comment (ckiee) [inline .github/workflows/backport.yml:8]:
```suggestion
# the GitHub repository. This means that it should not evaluate user input in a
```

**Review by ckiee — CHANGES_REQUESTED** [review]

File: .github/workflows/pending-set.yml [hunk 2]
```diff
@@ -3,6 +3,11 @@ name: "set pending status"
 on:
   pull_request_target:
 
+# WARNING:
+# When extending this action, be aware that $GITHUB_TOKEN allows write access to
+# the github repository. This means that is should not evaluate user input in a
```
Comment (Mic92) [inline .github/workflows/pending-set.yml:8]:
```suggestion
# the GitHub repository. This means that it should not evaluate user input in a
```

**Review by ckiee — APPROVED** [review]
Nice. Squash & merge?

**Comment by zowoq** [comment]
Might be missing something but why just these two, don't a couple of other actions also use a write token?

**Comment by ckiee** [comment]
I didn't really think about that since I just look at the diffs but it does seem there's actually a few that don't have `permissions: read-all` that get triggered `on: pull_request*`:

`ckie@cookiemonster ~/git/nixpkgs -> rg pull_request .github --files-with-matches | sed -e 's/^/- [ ] `/' | sed -e 's/$/`/'`
- [x] `.github/workflows/manual-nixpkgs.yml`
- [x] `.github/workflows/manual-nixos.yml`
- [ ] `.github/workflows/basic-eval.yml`
- [ ] `.github/workflows/no-channel.yml`
- [ ] `.github/workflows/labels.yml`
- [x] `.github/workflows/nixos-manual.yml`
- [x] `.github/workflows/editorconfig.yml`
- [x] `.github/workflows/pending-set.yml`
- [x] `.github/workflows/backport.yml`

**Comment by Mic92** [comment]
- .github/workflows/basic-eval.yml: does not use pull_request_target
- .github/workflows/no-channel.yml: same
- .github/workflows/labels.yml: Not sure about this one, it allows writing to the current pull-request but not content? Maybe I add a warning just in case.

**Review by zowoq — APPROVED** [review]

**Pull request merged**